### ISSUE TITLE
No such file or directory: 

### ISSUE THREAD

**Issue by moxxuk** [catalyst/moodle-cachestore_redissentinel#1]
I can't get this working, mainly due to a required file not being present in lib.php:

`require_once($CFG->libdir.'/redis/sentinel.php');`

Where does this sentinel.php file come from?

Many thanks

**Comment by danmarsden** [comment]
I'm not completely sure why that was dropped in lib/redis - but heres the relevant file:

[sentinel.zip](https://github.com/catalyst/moodle-cachestore_redissentinel/files/3769359/sentinel.zip)

Feel free to submit a pull request to move that into the plugin's own directory instead of in /lib/

**Comment by mrkskwsnck** [comment]
@danmarsden Is this zipped `sentinel.php` file part of this project or of an external origin?

**Comment by danmarsden** [comment]
@mrkskwsnck - it's code that Catalyst staff have written to support this plugin - it should have Moodle's standard GPL statement at the top along with the copyright assigned to Catalyst IT - it's a class required for this plugin to work.

**Comment by mrkskwsnck** [comment]
@danmarsden Unfortunately, the zipped file has neither a license statement nor any other comments. Also, I could not find this file in any other of the Catalyst projects :-( Because of that I would not recommend to commit it to this project by a PR. IMHO

**Comment by danmarsden** [comment]
Thanks - I can confirm that the file was written by catalyst staff(Matt clarkson) and although it is missing that detail in the zip it is appropriate to add them. it sits in our private local repository but was added here as a zip in case someone else wants to use this code.

at some point we may get round to tidying this up ourselves but a PR would speed it up.

**Comment by mrkskwsnck** [comment]
Now, that I have my Redis/Sentinel Cluster up and running using this cache store, I created the PR https://github.com/catalyst/moodle-cachestore_redissentinel/pull/3 containing the missing class. Also I explained some essential configuration details in the README file, which were not obvious for me.

PS: I would appreciate it, if you would choose a license for this project. Preferably GPL v3 to be in sync with Moodle.
